coins/monero/tests/send.rs

```diff
@@ -55,168 +53,305 @@ macro_rules! async_sequential {
-        let output = txs.swap_remove(0).ignore_timelock().swap_remove(0);
-        amount += output.commitment().amount;
-        outputs.push(output);
-      }
+    let mut scanner = Scanner::from_view(
+      ViewPair::new(spend_pub, view),
+      Network::Mainnet,
+      Some(HashSet::new())
+    );
+    let addr = scanner.address();
+    let fee = rpc.get_fee().await.unwrap();
+
+    // We actually need 120 decoys for this transaction, so mine until then
+    // 120 + 60 (miner TX maturity) + 10 (lock blocks)
+    // It is possible for this to be lower, by noting maturity is sufficient regardless of lock
+    // blocks, yet that's not currently implemented
+    // TODO, if we care
+    rpc.mine_regtest_blocks(&addr.to_string(), 200).await.unwrap();
+    let start = rpc.get_height().await.unwrap() - 68;
+
+    // spent 8 output
+    let mut outputs = vec![];
+    let mut amount = 0;
+    for i in (start + 1) .. (start + 9) {
+      let mut txs = scanner.scan(&rpc, &rpc.get_block(i).await.unwrap()).await.unwrap();
+      let output = txs.swap_remove(0).ignore_timelock().swap_remove(0);
+      amount += output.commitment().amount;
+      outputs.push(output);
     }
 
+    // make a tx and sign
     let mut signable = SignableTransaction::new(
       rpc.get_protocol().await.unwrap(),
       outputs,
-      vec![(addr, amount - 10000000000)],
+      vec![(addr, amount - 10_000_000_000)], // 0.01 xmr buffer for the tx fee
       Some(addr),
       None,
       fee,
     )
     .unwrap();
+    let tx = signable.sign(&mut OsRng, &rpc, &spend).await.unwrap();
 
-    if !multisig {
-      tx = Some(signable.sign(&mut OsRng, &rpc, &spend).await.unwrap());
-    } else {
-      #[cfg(feature = "multisig")]
-      {
-        let mut machines = HashMap::new();
-        for i in 1 ..= THRESHOLD {
-          machines.insert(
-            i,
-            signable
-              .clone()
-              .multisig(
-                &rpc,
-                keys[&i].clone(),
-                RecommendedTranscript::new(b"Monero Serai Test Transaction"),
-                rpc.get_height().await.unwrap() - 10,
-                (1 ..= THRESHOLD).collect::<Vec<_>>(),
-              )
-              .await
-              .unwrap(),
-          );
-        }
-
-        tx = Some(sign(&mut OsRng, machines, &vec![]));
-      }
-    }
-
-    rpc.publish_transaction(tx.as_ref().unwrap()).await.unwrap();
-    mine_block(&rpc, &addr.to_string()).await.unwrap();
-  }
-}
-
-async_sequential! {
-  async fn send_single_input() {
-    send_core(0, false).await;
-  }
-
-  async fn send_multiple_inputs() {
-    send_core(1, false).await;
+    // publish the tx
+    rpc.publish_transaction(&tx).await.unwrap();
   }
 }
 
 #[cfg(feature = "multisig")]
 async_sequential! {
   async fn multisig_send_single_input() {
-    send_core(0, true).await;
+    let rpc = Rpc::new("http://127.0.0.1:18081".to_string()).unwrap();
```

Comment:
This large block of additions and the block below it are largely duplicated code. While the point of this PR is to remove the monolithic function, duplicating it everywhere isn't a clean solution.

We may *need* to have these tests written around TransactionBuilder with a custom test runner to ensure cleanliness. To recap our previous discussions:
- Each test would take in a `&mut Builder` and perform operations
- The test runner would create a Builder, assign it a coinbase output as an input, and pass a builder to modify
- The test runner would sign and confirm it can be broadcasted
- The test runner would then call another function to have whatever desired checks performed (such as, the transaction can then be scanned)

This minimal runner design doesn't work in the long-term however as it's only designed for a single input. We either need:
1) A runner which can take the above design yet run multiple transactions. This would likely have to be a macro.
2) A very suitable set of helper functions combined with some statics.

I'll put forth a commit offering the former as it should be sufficiently brief and within my skill set.